Simplenote/src/main/java/com/automattic/simplenote/LastSyncTimeCache.java

```diff
@@ -0,0 +1,60 @@
+package com.automattic.simplenote;
+
+import com.simperium.client.Bucket;
+import com.simperium.client.Syncable;
+
+import java.util.Calendar;
+import java.util.HashMap;
+
+public class LastSyncTimeCache<T extends Syncable> {
+    private HashMap<String, Calendar> mSyncTimes = new HashMap<>();
+    private SyncTimeListener mListener;
+
+    public Calendar getLastSyncTime(String key) {
+        return mSyncTimes.get(key);
+    }
+
+    public void addListener(SyncTimeListener listener) {
+        mListener = listener;
+    }
+
+    private void updateSyncTime(String entityId) {
+        Calendar now = Calendar.getInstance();
+        mSyncTimes.put(entityId, now);
+
+        if (null != mListener) {
+            mListener.onUpdate(entityId, now);
+        }
+    }
+
+    public Bucket.Listener<T> listener = new Bucket.Listener<T>() {
+        @Override
+        public void onSaveObject(Bucket<T> bucket, T object) {
+            updateSyncTime(object.getSimperiumKey());
```

Comment:
@theck13 this is the line I'm guessing is causing the sync time to update when offline.

originally I didn't have this here but then I wasn't getting a time update when making changes on the device. that is, we're updating when we receive updates from the server but we're not listening for the acknowledged change when the patch originated locally.

I'll do more digging and see if I can find another way to get that info

Reply:
> we're updating when we receive updates from the server but we're not listening for the acknowledged change when the patch originated locally.

That's exactly right.  `onSaveObject` is triggered when an object is saved in Simperium Android _locally_.  `onNetworkChange` is triggered when an object is updated _remotely_.